common4j/src/main/com/microsoft/identity/common/java/cache/MsalOAuth2TokenCache.java

```diff
@@ -1661,7 +1661,8 @@ private void deleteAccessTokensWithIntersectingScopes(
                 referenceToken.getRealm(),
                 null, // Wildcard (*)
                 referenceToken.getAccessTokenType(),
-                referenceToken.getRequestedClaims()
+                referenceToken.getRequestedClaims(),
+                mAccountCredentialCache.getCredentials()
```

Comment:
>mAccountCredentialCache.getCredentials() [](http://example.com/codeflow?start=16&length=40)

This call is still passing all credentials from same object (effectively same as before). 

From your description, do you mean the benefit would be when there are multiple calls to getCredentialsFilteredBy, same list can be passed to each, avoiding internal call to getCredentials?

Reply:
Yes, that's exactly right.  This particular one doesn't benefit -- it's okay because all the paths that call into this function only call it once, and don't call anything else that invokes 'getCredentials'.

Reply:
The usage of this will come on the OneAuth/MSAL side.  We have a read method that loops over each credential type and calls `getCredentialsFilteredBy` each time.  Since that internally calls `getCredentials`, it winds up getting called many times.  With this new overload, I can call `getCredentials` once at the top of the function, and reuse it for each filter call (which is fast).